pkg/controller/core/localqueue_controller.go

```diff
@@ -174,11 +194,104 @@ func (h *qWorkloadHandler) Generic(e event.GenericEvent, q workqueue.RateLimitin
 	q.AddAfter(req, constants.UpdatesBatchPeriod)
 }
 
+// qCQHandler signals the controller to reconcile the Queue associated
+// to the workload in the event.
+type qCQHandler struct {
+	client client.Client
+}
+
+func (h *qCQHandler) Create(e event.CreateEvent, wq workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	h.addLocalQueueToWorkQueue(cq, wq)
+}
+
+func (h *qCQHandler) Update(e event.UpdateEvent, wq workqueue.RateLimitingInterface) {
+	newCq, ok := e.ObjectNew.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	oldCq, ok := e.ObjectOld.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	// Iff .status.conditions of the clusterQueue is updated,
+	// this handler sends all queues related to the clusterQueue to workqueue.
+	if equality.Semantic.DeepEqual(oldCq.Status.Conditions, newCq.Status.Conditions) {
+		return
+	}
+	h.addLocalQueueToWorkQueue(newCq, wq)
+}
+
+func (h *qCQHandler) Delete(e event.DeleteEvent, wq workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	h.addLocalQueueToWorkQueue(cq, wq)
+}
+
+func (h *qCQHandler) Generic(event.GenericEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) addLocalQueueToWorkQueue(cq *kueue.ClusterQueue, wq workqueue.RateLimitingInterface) {
+	ctx := context.TODO()
+	log := ctrl.LoggerFrom(ctx).WithValues("clusterQueue", klog.KObj(cq))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	var queues kueue.LocalQueueList
+	err := h.client.List(ctx, &queues, client.MatchingFields{indexer.QueueClusterQueueKey: cq.Name})
+	if err != nil {
+		log.Error(err, "listing queues that match the clusterQueue")
+		return
+	}
+	for _, q := range queues.Items {
+		req := reconcile.Request{
+			NamespacedName: client.ObjectKeyFromObject(&q),
+		}
+		wq.AddAfter(req, constants.UpdatesBatchPeriod)
```

Comment:
I don't expect these updates to happen often, so no need for batching, but also ok to leave.

Reply:
> I don't expect these updates to happen often, so no need for batching

Makes sense. I will modify it so that a queue is added to the workqueue, immediately.

Reply:
Done.